packages/jsii-pacmak/lib/targets/go/types/go-type-reference.ts

```diff
@@ -72,6 +72,11 @@ export class GoTypeRef {
     return this.type?.name;
   }
 
+  public get datatype() {
+    const reflectType = this.type?.type;
+    return reflectType?.isInterfaceType() && reflectType?.datatype;
```

Comment:
I'd assume the compiler can infer the following is correct... Can it not?

```suggestion
    return reflectType?.isInterfaceType() && reflectType.datatype;
```